machine-learning-box/customer-lifetime-value/README.md

```diff
@@ -0,0 +1,74 @@
+Customer Lifetime Value Prediction Template
+===
+
+Predicting Customer Lifetime Value (CLTV) allows you to efficiently identify high-value customers and understand their characteristics. Eventually, you can effectively optimize your day-to-day marketing activities without missing important highly-engaged customers.
+
+## Input
+
+As an online retailer, assume we have [a `source` table](./config/general.yml#L3) that represents customer's order histories as follows:
+
+| InvoiceNo<br/>`string` | InvoiceDate<br/>`string` | CustomerID<br/>`long` | Country<br/>`string` | StockCode<br/>`string` | Description<br/>`string` | UnitPrice <br/>`double` | Quantity<br/>`long` | 
+|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| 536365 | 2010/12/01 8:26 | 17850 | United Kingdom | 85123A | WHITE HANGING HEART T-LIGHT HOLDER | 2.55 | 6 |
+| 536365 | 2010/12/01 8:26 | 17850 | United Kingdom | 71053 | WHITE METAL LANTERN | 3.39 | 6 |
+| 536365 | 2010/12/01 8:26 | 17850 | United Kingdom | 84406B | CREAM CUPID HEARTS COAT HANGER | 2.75 | 8 |
+| ... |...|...| ...|...|...|...|...|
+| C536391 | 2010/12/01 10:24 | 17548 | United Kingdom | 22556 | PLASTERS IN TIN CIRCUS PARADE  | 1.65 | -12 |
+| ... |...|...| ...|...|...|...|...|
+
+Here, each row represents single item (i.e., **StockCode**) in a basket corresponding to single order (i.e., **InvoiceNo**). That is, a CLTV can be calculated by **a sum of (UnitPrice * Quantity)** for unique CustomerID.
+
+> **Note:** Negative **Quantity** represents returned items.
+
+## Workflow
+
+First, prepare the data as follows:
+
+1. Download `Online Retail.xlsx` from [UCI Machine Learning Repository: Online Retail Data Set](http://archive.ics.uci.edu/ml/datasets/Online+Retail). Our data model explained above is fully based on this public dataset.
+2. Convert the data into a CSV file.
+3. Import the CSV file to a Treasure Data table. 
```

Comment:
Not mandatory, but how about adding ingestion workflow with py> operator and `pandas.read_excel`?

Reply:
I like that idea! 239e3296fde89ec8f45a4935ac30258c92402440